#!/usr/bin/env python

PACKAGE = "gr_map_utils"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("map_resolution",  double_t,   0, "Map resolution", 0.025, 0.0, 5.0)
gen.add("map_offset",  double_t,   0, "Map offset to origin", 2.0, 0.0, 150.0)
gen.add("constrain_motion",  bool_t,   0, "Inverted Free/Occupied", True)
gen.add("mark_nodes",  bool_t,   0, "Mark Nodes", True)
gen.add("mark_edges",  bool_t,   0, "Mark Edges", True)
gen.add("feature_inflation",  int_t,   0, "Value on Costmap Nodes", 10, 0, 100)
gen.add("nodes_value",  int_t,   0, "Nodes Value on Costmap", 0, 0, 255)
gen.add("edges_value",  int_t,   0, "Edges Value on Costmap", 0, 0, 255)

gen.add("map_orientation",  double_t,   0, "Map angle (map frame)", 0, -1.575, 1.575)

exit(gen.generate(PACKAGE, "gr_map_utils", "TopologicalMapConverter"))
